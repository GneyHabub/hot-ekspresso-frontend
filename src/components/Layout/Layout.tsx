import { Box } from '@mui/system';
import React from 'react';
import * as colors from '@mui/material/colors';
import {
  AppBar, Container, IconButton, Link, Menu, MenuItem, Toolbar, Typography,
} from '@mui/material';
import { useAuth0 } from '@auth0/auth0-react';
import AccountCircle from '@mui/icons-material/AccountCircle';
import MenuIcon from '@mui/icons-material/Menu';

interface Props {
  onSidebarOpen?: () => void
}

const Layout: React.FC<Props> = ({ children, onSidebarOpen }) => {
  const { isAuthenticated, logout } = useAuth0();
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);

  const handleMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };
  return (
    <Box
      sx={{
        width: '100%',
        height: '100%',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        rowGap: '30px',
        backgroundColor: colors.grey[50],
        overflowY: 'scroll',
      }}
    >
      <AppBar position="static">
        <Toolbar>
          {isAuthenticated
            && (
            <IconButton
              size="large"
              edge="start"
              color="inherit"
              aria-label="menu"
              sx={{ mr: 2 }}
              onClick={onSidebarOpen}
            >
              <MenuIcon />
            </IconButton>
            )}
          <Link
            sx={{
              flexGrow: 1,
              cursor: 'pointer',
            }}
            href="/"
            variant="h6"
            component="div"
            underline="none"
            color="white"
          >
            Hot Ekspresso Travel ✈️
          </Link>
          {isAuthenticated
            && (
            <div>
              <IconButton
                size="large"
                aria-label="account of current user"
                aria-controls="menu-appbar"
                aria-haspopup="true"
                onClick={handleMenu}
                color="inherit"
              >
                <AccountCircle />
              </IconButton>
              <Menu
                id="menu-appbar"
                anchorEl={anchorEl}
                anchorOrigin={{
                  vertical: 'top',
                  horizontal: 'right',
                }}
                keepMounted
                transformOrigin={{
                  vertical: 'top',
                  horizontal: 'right',
                }}
                open={Boolean(anchorEl)}
                onClose={handleClose}
              >
                <MenuItem onClick={() => logout()}>Logout</MenuItem>
              </Menu>
            </div>
            )}
        </Toolbar>
      </AppBar>
      <Container
        maxWidth="lg"
      >
        {children}
      </Container>

    </Box>
  );
};

export default Layout;
