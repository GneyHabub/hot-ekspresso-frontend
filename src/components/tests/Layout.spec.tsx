import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import Layout from '../Layout/Layout';

describe('Login', () => {
  it('matches the snapshot', () => {
    const wrapper = render(<BrowserRouter><Layout /></BrowserRouter>);
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
