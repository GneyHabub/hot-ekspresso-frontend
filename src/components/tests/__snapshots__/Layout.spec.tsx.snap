// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1yg6i26"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          
          <div
            class="MuiTypography-root MuiTypography-h6 MuiLink-root MuiLink-underlineNone css-1qqbuje-MuiTypography-root-MuiLink-root"
            href="/"
          >
            Hot Ekspresso Travel ✈️
          </div>
          
        </div>
      </header>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      />
    </div>
  </div>
</body>
`;
